bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/models/Teaser.java

```diff
@@ -24,70 +24,130 @@
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
      */
+    String PN_HIDE_IMAGE_LINK = "hideImageLink";
 
-    String PN_LINK_TEXT = "linkText";
+    /**
+     * Name of the property that will store if the title should be hidden.
+     *
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    String PN_HIDE_TITLE = "hideTitle";
 
     /**
-     * Returns this teaser's title, if one was defined.
+     * Name of the property that will store if the title link should be hidden.
      *
-     * @return the teaser's title or {@code null}
-     * @since com.adobe.cq.wcm.core.components.sandbox.models 2.3.0
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
      */
-    default String getTitle() {
+    String PN_HIDE_TITLE_LINK = "hideTitleLink";
+
+    /**
+     * Name of the property that will store if the title value should be taken from the linked page.
+     *
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    String PN_TITLE_VALUE_FROM_PAGE = "titleValueFromPage";
+
+    /**
+     * Name of the property that will store if the description should be hidden.
+     *
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    String PN_HIDE_DESCRIPTION = "hideDescription";
+
+    /**
+     * Name of the property that will store if the description link should be hidden.
+     *
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    String PN_HIDE_DESCRIPTION_LINK = "hideDescriptionLink";
+
+    /**
+     * Name of the property that will store if the description value should be taken from the linked page.
+     *
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    String PN_DESCRIPTION_VALUE_FROM_PAGE = "descriptionValueFromPage";
+
+    /**
+     * Returns the URL to which this teaser links, if one was defined.
+     *
+     * @return the URL to which teaser links or {@code null}
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    default String getLinkURL() {
         throw new UnsupportedOperationException();
     }
 
     /**
-     * Returns this teaser's description, if one was defined.
+     * Returns the image resource for this teaser.
+     * @return the image resource for this teaser or {@code null}
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
+     */
+    default Resource getImageResource() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Checks if link should be hidden on the image.
      *
-     * @return the teaser's description or {@code null}
-     * @since com.adobe.cq.wcm.core.components.sandbox.models 2.3.0
+     * @return {@code true} if link should be hidden on the image, {@code false} otherwise
+     * @since com.adobe.cq.wcm.core.components.sandbox.models 1.0.0
      */
-    default String getDescription() {
+    default boolean isHideImageLink() {
```

Comment:
I'd call it `hideImageLink`

Reply:
I prefer @vladbailescu 's approach, `is...` is a convention for `boolean`
`hideImageLink` sounds more like an action

Reply:
it's just that isHide... does not sound well. But I guess that the convention `is<property-name>` is fine. We could also rename the property `imageLinkHidden`. The method name `isImageLinkHidden` would then sound better. I'm fine with both.